### PR TITLE
feat(editor/ied): Filter for logical nodes

### LINKED ISSUE #591: Provide manufacturer and type filtering in IED Editor
Here's a use case for providing name, manufacturer and type filtering in the IED editor (with some kind of auto-completion).

1. We're evaluating some IEDs.
2. We said, "We'd quite like to have an SCD file with some mappings in it to look at". It turns out this device will only offer proprietary file formats or an IID file once loaded to the device (and we don't have one physically handy).
3. We received a link to a [large collection of SCD files](https://iec61850.ucaiug.org/2019IOP-Charlotte/IOP%20Test%20Documents/Forms/AllItems.aspx?RootFolder=%2f2019IOP-Charlotte%2fIOP%20Test%20Documents%2fSCD%20Files&FolderCTID=&View=%7BEC1FD7D0-5289-4545-86ED-2CC3926EA317%7D) from Interoperability Testing.
4. I thought "I'll choose one and find the right device by opening it in OpenSCD".
5. The IED editor seems to be the right place.
6. In the file I chose, not all the naming makes much sense to me and not all the manufacturers have included the manufacturer in the device description.
7. It would be useful if I had a drop-down to see values and filter on manufacturer and device type in the IED editor and to have the ability to free-text filter on these fields (name, manufacturer and device type).
8. I know what I'm looking for but I can't get there...

### LINKED ISSUE #856: Filter option for logical nodes inside the IED browser
**Is your feature request related to a problem? Please describe.**
As a user of OpenSCD I want to filter for display specific logical nodes. 

**Describe the solution you'd like**
Most data models become bigger and bigger. Therefore, filtering is key here. One of the most common filtering interests are function aka logical nodes. Being able to select multiple logical nodes that are display within the and the unselected are not displayed.

**Requirements**
- allow filter for logical node class as well as description when nsdoc description of available
- allow multiple selection
- allow select all (default)
- do not display leaf/empty (by filtering) logical devices 

**Suggestions for a specific user interaction**
logical_node_filter_mock excalidraw

### PR DESCRIPTION
**Pull request by dlabordus**
closes #591
closes #856

### CODE REVIEW

File: src/oscd-filter-button.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  private onClosing(): void {
+    const selectedItems: string[] = [];
+    if (this.selected) {
+      if (this.selected instanceof Array) {
+        this.selected.forEach(item => selectedItems.push(item.value));
+        this.dispatchEvent(newSelectedItemsChangedEvent(selectedItems));
+      } else {
+        selectedItems.push(this.selected.value);
+        this.dispatchEvent(newSelectedItemsChangedEvent(selectedItems));
```
Comment (pascalwilbrink) [inline src/oscd-filter-button.ts:45]: this.dispatchEvent(newSelectedItemsChangedEvent(selectedItems));

can be outside of the if/else statement (1 line can be removed)

File: src/editors/IED.ts [hunk 2]
```diff
@@ -56,60 +53,131 @@ export default class IedPlugin extends LitElement {
...
 
   @state()
-  private set selectedIed(element: Element | undefined) {
-    iedEditorSelectedIed = element;
+  private get lnClassList(): string[][] {
+    const currentIed = this.selectedIed;
+    const uniqueLNClassList: string[] = [];
+    if (currentIed) {
+      return Array.from(currentIed.querySelectorAll('LN0, LN'))
+        .filter(element => element.getAttribute('lnClass'))
```
Comment (pascalwilbrink) [inline src/editors/IED.ts:61]: can be ```element.hasAttribute('lnClass')``` ?

File: src/editors/IED.ts [hunk 3]
```diff
@@ -56,60 +53,131 @@ export default class IedPlugin extends LitElement {
...
-      iedEditorSelectedIed === undefined ||
-      iedEditorSelectedIed.parentElement === null
-    ) {
-      const iedList = this.alphabeticOrderedIeds;
-      if (iedList.length > 0) {
-        iedEditorSelectedIed = iedList[0];
-      }
+    if (this.selectedIEDs.length >= 1) {
+      const iedList = this.iedList;
+      return iedList.filter(element => {
```
Comment (pascalwilbrink) [inline src/editors/IED.ts:85]:
```
return iedList.find(element => {
   const iedName = getNameAttribute(element);
   return this.selectedIEDs[0] === iedName;
});
```

returns the first found

File: src/oscd-filter-button.ts (hunk 1 above)
Comment (dlabordus) [reply src/oscd-filter-button.ts]: Fixed

File: src/editors/IED.ts (hunk 2 above)
Comment (dlabordus) [reply src/editors/IED.ts]: Fixed

File: src/editors/IED.ts (hunk 3 above)
Comment (dlabordus) [reply src/editors/IED.ts]: Fixed

**Review by pascalwilbrink — APPROVED** [review]

**Review by JakobVogelsang — APPROVED** [review]
LGTM thx

**Pull request merged**